### PR TITLE
fix: Conceptual topic on dynamic binding

### PR DESCRIPTION
**Pull request by rashmi-sahoo-git**
## Checklist
I have:
- [x] run the content through Grammarly
- [ ] linked to sample apps when relevant
- [x] added the meta description for each page in the PR
- [x] minimized the callouts and added only when necessary
- [ ] added the `queryString` parameter to the Tabs (if used)
- [ ] masked PII in images. For example, login credentials, account details, and more
- [ ] added images only when necessary
- [ ] deleted the images that are no longer used for the updated pages in the PR
- [ ] followed the image file naming convention while renaming or adding new images. (Use lowercase letters, dashes between words, and be as descriptive as possible)
- [ ] used the `<figure/>` tag instead of a markdown representation for images
- [ ] added the `<figcaption/>` tag to add a caption to the image
- [ ] added the `alt` attribute in the `` tag
- [ ] verified and updated the cross-references or created redirect rules for the changed or removed content 
- [x] reviewed and applied the style changes for UI formatting. For example, Bold the UI elements(Buttons on screen) used in the doc.

### CODE REVIEW

File: website/docs/connect-data/concepts/dynamic-binding-in-queries.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+```js
+https://mock-api.appsmith.com/v1?{{<indexid.selectedOptionValue>}}
+```
+
+#### Headers
+
+You can dynamically add request headers to your API requests using dynamic binding.
+For example, you have a REST API `https://mock-api.appsmith.com`, and a [Select widget](https://docs.appsmith.com/reference/widgets/select) named `applicationid`. 
+
+<img src="/img/dynamic-binding-api-headers.png" style= {{width:"700px", height:"auto"}} alt="Dynamic binding in API headers"/>
```
Comment (jnikhila) [inline website/docs/connect-data/concepts/dynamic-binding-in-queries.md:58]: Use <figcaption> here.

**Review by jnikhila — CHANGES_REQUESTED** [review]

File: website/docs/connect-data/concepts/dynamic-binding-in-queries.md (hunk 1 above)
Comment (rashmi-sahoo-git) [reply website/docs/connect-data/concepts/dynamic-binding-in-queries.md]: Done.

File: website/docs/connect-data/concepts/dynamic-binding-in-queries.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+For example, you have a SQL data source with a `users` table, and a [Select widget](https://docs.appsmith.com/reference/widgets/select) named `userid`.
+
+Use the following syntax to filter the data for the selected user id:
+
+```js
+select id, name, email, country
+from users u where u.id = ({{userid.selectedOptionValue}})
+```
+
+### API queries
```
[the remark this answers is not in the document]
Comment (rashmi-sahoo-git) [reply website/docs/connect-data/concepts/dynamic-binding-in-queries.md:24]: Done.

File: website/docs/connect-data/concepts/dynamic-binding-in-queries.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+```
+
+Dynamic binding also lets you construct JSON objects dynamically to include in your request bodies.
+
+You can use dynamic binding with the following:
+- URI path
+- Query parameters
+- Headers
+- Body
+
```
Comment (jnikhila) [inline website/docs/connect-data/concepts/dynamic-binding-in-queries.md:43]: Please add one generic example of the binding here.

**Review by jnikhila — CHANGES_REQUESTED** [review]

File: website/docs/connect-data/concepts/README.md [hunk 4]
```diff
@@ -1,37 +1,51 @@
 # Concepts
 
 <div class="containerGridSampleApp">
-  <div class="containerColumnSampleApp columnGrid column-one">
+<div class="containerColumnSampleApp columnGrid column-one">
+        <div class="containerCol">
+            <a href="/connect-data/concepts/dynamic-binding-in-queries"><strong>Dynamic Binding in Queries</strong></a>
+        </div> <hr/>
+        <div class="containerDescription">Bind data dynamically to your queries in Appsmith.</div>
```
Comment (jnikhila) [inline website/docs/connect-data/concepts/README.md:8]: Can you elaborate and come up with a better description?

Comment (rashmi-sahoo-git) [reply website/docs/connect-data/concepts/README.md]: Done.

**Review by jnikhila — APPROVED** [review]

**Pull request merged**